packages/dashboard-v2/src/components/AvatarUploader/AvatarUploader.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+
+import { useUser } from "../../contexts/user";
+import { SimpleUploadIcon } from "../Icons";
+
+const AVATAR_PLACEHOLDER = "/images/avatar-placeholder.svg";
+
+export const AvatarUploader = (props) => {
+  const { user } = useUser();
+  const [imageUrl, setImageUrl] = useState(AVATAR_PLACEHOLDER);
+
+  useEffect(() => {
+    setImageUrl(user.avatarUrl ?? AVATAR_PLACEHOLDER);
+  }, [user]);
+
+  return (
+    <div {...props}>
+      <div
+        className={`flex justify-center items-center xl:w-[245px] xl:h-[245px] bg-contain bg-none xl:bg-[url(/images/avatar-bg.svg)]`}
+      >
+        <img src={imageUrl} className="w-[160px]" alt="" />
+      </div>
+      <div className="flex justify-center">
+        <button
+          className="flex items-center gap-4 hover:underline decoration-1 decoration-dashed underline-offset-2 decoration-gray-400"
+          type="button"
+          onClick={console.info.bind(console)}
+        >
+          <SimpleUploadIcon size={20} className="shrink-0" /> Upload profile picture
+        </button>
+        {/* TODO: actual uploading */}
```

Comment:
note: we might want to hold off on that feature, not sure where we would upload this since servers do not share files and uploading to skynet from browser might be problematic (user might not have active subscription) unless we have a dedicated account service endpoint that uploads and allows to download the image without the need for the subscription. We might need to create internal upload/download endpoints or lift account restrictions based on the source ip.

Reply:
👍